### PR TITLE
feat: state management (CORE-5560)

### PR DESCRIPTION
**Pull request by leartgjoni-voiceflow**
# State Management APIs

description: wrapper around the existing `/interact` endpoint that adds a data layer for storing the user's session.

⚠️ ATTENTION ⚠️ : this PR requires this [migration](https://github.com/voiceflow/database-cli/pull/60) to be merged and run beforehand
--

⚠️ ATTENTION ⚠️ : this PR requires the following env vars before deploying
--
```
SESSIONS_SOURCE='mongo'
MONGO_URI
MONGO_DB
```

⚠️ ATTENTION ⚠️ : after merging this PR, the [new updated runtime-client-sdk](https://github.com/voiceflow/runtime-client-js/pull/55) should be published as well
--

Testing
--
- postman collection and vf files to test endpoints:
[state management.zip](https://github.com/voiceflow/general-runtime/files/6245882/state.management.zip)

### CODE REVIEW

File: backend/api/routers/state.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const router = express.Router();
+
+  router.use(bodyParser.json({ limit: BODY_PARSER_SIZE_LIMIT }));
+  router.use(middlewares.rateLimit.verify);
+
+  router.post('/:versionID/user/:userID/interact', middlewares.rateLimit.consume, controllers.stateManagement.interact);
+
+  router.get('/:versionID/user/:userID', middlewares.rateLimit.consume, middlewares.version.hasPermission, controllers.stateManagement.get);
+
+  router.put('/:versionID/user/:userID', middlewares.rateLimit.consume, middlewares.version.hasPermission, controllers.stateManagement.update);
```
Comment (leartgjoni-voiceflow) [inline backend/api/routers/state.ts:17]: separately check permissions on these 2 endpoints. the other 2 already have permission checking built in by re-using the `interact` service.

File: lib/services/interact.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { Config } from '@voiceflow/general-types';
```
Comment (leartgjoni-voiceflow) [inline lib/services/interact.ts:1]: moved most of this logic from the controller to the service, to make it re-usable by both `interact` and `state` controllers

File: lib/services/session/index.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { State } from '@voiceflow/runtime';
```
Comment (leartgjoni-voiceflow) [inline lib/services/session/index.ts:1]: re-used the same patterns we've used before to have flexibility on the storage option for the sessions. (here currently supported: local, mongo)

**Comment by DecathectZero** [comment]
added ticket for ops team https://www.notion.so/voiceflow/Add-env-var-general-runtime-aebe4bd5825f4c978b3695c8e1fd4ed8

**Pull request merged**